calico_containers/pycalico/datastore.py

```diff
@@ -134,7 +134,15 @@ class DatastoreClient(object):
 
     def __init__(self):
         etcd_authority = os.getenv(ETCD_AUTHORITY_ENV, ETCD_AUTHORITY_DEFAULT)
-        (host, port) = etcd_authority.split(":", 1)
+        try:
+            (host, port) = etcd_authority.split(":", 1)
```

Comment:
I think the most robust way to do this is to define a regex pattern which matches valid etcd authorities and parse it using that.

For example, this won't catch if someone puts in `http://host:port` as their etcd authorit (which is a common and easy mistake for users to make)